proj.sh

```diff
@@ -2,6 +2,7 @@
 export PATH=$PATH:/usr/local/bin
 LOCALENV="local"
 METATLAS_DEFAULT_ENV="env-${LOCALENV}.env"
+export CHOSEN_ENV=$METATLAS_DEFAULT_ENV
```

Comment:
If `CHOSEN_ENV` is set before `proj.sh` is read, the variable will be overwritten by this line.
```sh
$ export CHOSEN_ENV=env-dev.env
$ source proj.sh
$ echo $CHOSEN_ENV 
env-local.env
```

Something like this would avoid that:
```sh
if [ -z "$CHOSEN_ENV" ];
then
  export CHOSEN_ENV=$METATLAS_DEFAULT_ENV
fi
```

Reply:
Maybe this is a bit overkill, but it could be nice to have function to set the `CHOSEN_ENV`, so that you can easily change it:
```sh
$ set-env dev
Using env-dev.env
```
The code:
```sh
function set-env {
  export CHOSEN_ENV="env-${1}.env"
  echo "Using $CHOSEN_ENV"
}
```

Reply:
This is a good catch @MalinAhlberg! Just to double check, I think the problem you are illustrating could be problematic for when deploying right? Since the current first line in the `deploy-stack` function only sets the `CHOSEN_ENV` but does not export it, it would mean that the docker compose files could end up with the wrong `CHOSEN_ENV`?

I like your `set-env` function and I think if we use that whenever we want to set the `CHOSEN_ENV`, on line 5 and inside the `deploy-stack` function, then we should be safe and would not need to have the `if - then` statement?

Reply:
I think it would still overwrite the old env variable when reading `proj.sh`, no? The code below would keep the old value (but is weird..?):
```sh
function set-env {
  export CHOSEN_ENV="env-${1-LOCALENV}.env"
  echo "Using $CHOSEN_ENV"
}

# if not set, set chosen_env to the default (localenv)
set-env=$CHOSEN_ENV
```
But the more I look at `proj.sh`, the more unsure I get of how we really want it to work.  Since the `CHOSEN_ENV` is exported, I think it should be possible to set it from outside of this script, and still keep that variable set when reading/loading `proj.sh`. So you should be able to go
```sh
export CHOSEN_ENV=another_env
source proj.sh
build-stack
````
and expect docker compose to work with `another_env`? Or maybe `deploy-stack` is the only place where the env should be able to differ from `local`?

Reply:
I think it would still overwrite the old en variable but I think that's ok.

The env is set globally on line 5 and this should be the default when working locally.

When deploying, the env should be set to the target environment (`dev` or `prod`) at the beginning and reset back to the default (`local`) at the end.

In other words, the only time when the env should be something other than `local` is during deploy. So I think it's ok to overwrite the env globally whenever we source the `proj.sh` file. Does that answer your question @MalinAhlberg ?

Reply:
Yes, thanks for the explanation! I guess it will work well for all our intentions, then :+1:

Reply:
I've added my two cents [below](https://github.com/MetabolicAtlas/MetabolicAtlas/pull/825#issuecomment-1010146492).

Reply:
I think this thread has been resolved now, please `unresolve` if otherwise.